addons/t9n/models/message.py

```diff
@@ -11,14 +11,67 @@ class Message(models.Model):
     _description = "Localizable message"
 
     body = fields.Text(
+        "Entry to be Translated",
         help="The actual, textual content to be translated.",
     )
+    context = fields.Char(
+        help="Context of the message.",
+    )
+    translator_comments = fields.Text(
+        help="Comments wrote by the translator/developer in the resource file.",
+    )
+    extracted_comments = fields.Text(
+        help="Comments extracted during the process of extracting translatable strings.",
+    )
+    references = fields.Text(
+        help="The full text that represents the references, one per line.",
+    )
     resource_id = fields.Many2one(
         comodel_name="t9n.resource",
         help="The resource (typically a file) from which the entry is coming from.",
+        ondelete="cascade",
+        required=True,
     )
     translation_ids = fields.One2many(
         comodel_name="t9n.translation",
         inverse_name="source_id",
         string="Translations",
     )
+
+    _sql_constraints = [
+        (
+            "body_context_resource_unique",
+            "UNIQUE(body, context, resource_id)",
+            "The combination of the message body and context must be unique within the same resource!"
+        ),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        updated_records = self.env[self._name]
```

Comment:
is there any reason not to write the name explicily?
```suggestion
        updated_records = self.env["t9n.message"]
```

Reply:
no there is not. your changes make more sense.